lib/index.js

```diff
@@ -507,14 +527,13 @@ Socket.prototype._flush = function() {
     };
 
     if (flags & zmq.ZMQ_POLLIN) {
-      emitArgs = ['message'];
+      emitArgs = this._read();
 
```

Comment:
Socket is not related to/inheriting from stream, check the implementation of ReadableStream for that.